src/masternode-sync.cpp

```diff
@@ -295,6 +295,27 @@ void CMasternodeSync::ProcessTick()
                 continue;
             }
 
+            CNodeStateStats stats;
+            if(!GetNodeStateStats(pnode->id, stats) || stats.nCommonHeight == -1 || stats.nSyncHeight == -1) continue; // not enough info about this peer
+
+            if(pCurrentBlockIndex->nHeight > stats.nCommonHeight) {
```

Comment:
Maybe it would be good to leave a little margin of error here.  If a new block just came in one node might receive it a little before the other.